### PR TITLE
Add information about uiRegistry component searching by properties

### PR DESCRIPTION
**Pull request by serhiyzhovnir**
## Purpose of this PR

This PR adds information about how to search the JS component via uiRegistry by some property.

## Affected URLs

- https://devdocs.magento.com/guides/v2.3/ui_comp_guide/concepts/ui_comp_uiregistry.html
- https://devdocs.magento.com/guides/v2.2/ui_comp_guide/concepts/ui_comp_uiregistry.html
- https://devdocs.magento.com/guides/v2.1/ui_comp_guide/concepts/ui_comp_uiregistry.html

whatsnew
Added a [new code example](https://devdocs.magento.com/guides/v2.1/ui_comp_guide/concepts/ui_comp_uiregistry.html#examples) that shows how to search for a component based on a property value using a uiRegistry instance

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.1/ui_comp_guide/concepts/ui_comp_uiregistry.md [hunk 1]
```diff
@@ -24,13 +24,33 @@ Now we have `uiRegistry` instance in the `registry` variable. We can use it to g
...
+The `uiRegistry` instance allows to search the component by some property.
+If you know some unique property of the component that you need to find, you can use the following code to get this component:
+
+```javascript
+var component = registry.get('%property% = %propertyValue%');
+```
+
+**Example**
+
+Example with getting component by full component name:
```
Comment (rogyar) [inline guides/v2.1/ui_comp_guide/concepts/ui_comp_uiregistry.md]: Please, rephrase as "Getting a component by the full name:" (remove "Example with"). This section is already wrapped by "Example" title.

File: guides/v2.1/ui_comp_guide/concepts/ui_comp_uiregistry.md [hunk 2]
```diff
@@ -24,13 +24,33 @@ Now we have `uiRegistry` instance in the `registry` variable. We can use it to g
...
 
 ```javascript
 // Admin > Products > Catalog > Add Product
 var fieldName = registry.get('product_form.product_form.product-details.container_name.name');
 ```
 
+Example with getting component by some property value:
```
Comment (rogyar) [inline guides/v2.1/ui_comp_guide/concepts/ui_comp_uiregistry.md]: Please, change here as well

File: guides/v2.1/ui_comp_guide/concepts/ui_comp_uiregistry.md [hunk 3]
```diff
@@ -24,13 +24,33 @@ Now we have `uiRegistry` instance in the `registry` variable. We can use it to g
 var component = registry.get('%componentName%');
 ```
 
-For example:
+The `uiRegistry` instance allows to search the component by some property.
```
Comment (jeff-matthews) [inline guides/v2.1/ui_comp_guide/concepts/ui_comp_uiregistry.md]:
```suggestion
The `uiRegistry` instance allows you to search for components using property values.
```

File: guides/v2.1/ui_comp_guide/concepts/ui_comp_uiregistry.md [hunk 4]
```diff
@@ -24,13 +24,33 @@ Now we have `uiRegistry` instance in the `registry` variable. We can use it to g
...
+If you know some unique property of the component that you need to find, you can use the following code to get this component:
```
Comment (jeff-matthews) [inline guides/v2.1/ui_comp_guide/concepts/ui_comp_uiregistry.md]:
```suggestion
If you know a unique property value of a component that you need to find, you can use the following code to get the component:
```

File: guides/v2.1/ui_comp_guide/concepts/ui_comp_uiregistry.md [hunk 5]
```diff
@@ -24,13 +24,33 @@ Now we have `uiRegistry` instance in the `registry` variable. We can use it to g
...
+
+```javascript
+var component = registry.get('%property% = %propertyValue%');
+```
+
+**Example of usages**
```
Comment (jeff-matthews) [inline guides/v2.1/ui_comp_guide/concepts/ui_comp_uiregistry.md]:
```suggestion
**Examples**
```

File: guides/v2.1/ui_comp_guide/concepts/ui_comp_uiregistry.md [hunk 6]
```diff
@@ -24,13 +24,33 @@ Now we have `uiRegistry` instance in the `registry` variable. We can use it to g
...
+
+The code below lists how to get a component by a full component name:
```
Comment (jeff-matthews) [inline guides/v2.1/ui_comp_guide/concepts/ui_comp_uiregistry.md]:
```suggestion
The following code shows how to get a component by a full component name:
```

File: guides/v2.1/ui_comp_guide/concepts/ui_comp_uiregistry.md [hunk 7]
```diff
@@ -24,13 +24,33 @@ Now we have `uiRegistry` instance in the `registry` variable. We can use it to g
...
 
 ```javascript
 // Admin > Products > Catalog > Add Product
 var fieldName = registry.get('product_form.product_form.product-details.container_name.name');
 ```
 
+The code below lists how to get a component by some property value:
```
Comment (jeff-matthews) [inline guides/v2.1/ui_comp_guide/concepts/ui_comp_uiregistry.md]:
```suggestion
The following code shows how to get a component by a property value:
```

File: guides/v2.1/ui_comp_guide/concepts/ui_comp_uiregistry.md [hunk 8]
```diff
@@ -24,13 +24,33 @@ Now we have `uiRegistry` instance in the `registry` variable. We can use it to g
...
+
+```javascript
+// Admin > Products > Catalog > Add Product
+var fieldName = registry.get('index = name');
+// or 
+fieldName = registry.get('inputName = product[name]');
+```
+
+By this code, you will obtain the same JS component as with using the full component name.
```
Comment (jeff-matthews) [inline guides/v2.1/ui_comp_guide/concepts/ui_comp_uiregistry.md]:
```suggestion
The previous example gets the same JS component as using the full component name.
```

File: guides/v2.2/ui_comp_guide/concepts/ui_comp_uiregistry.md [hunk 9]
```diff
@@ -24,13 +24,33 @@ Now we have `uiRegistry` instance in the `registry` variable. We can use it to g
 var component = registry.get('%componentName%');
 ```
 
-For example:
+The `uiRegistry` instance allows to search the component by some property.
```
Comment (jeff-matthews) [inline guides/v2.2/ui_comp_guide/concepts/ui_comp_uiregistry.md]:
```suggestion
The `uiRegistry` instance allows you to search for components by property values. 
```

File: guides/v2.2/ui_comp_guide/concepts/ui_comp_uiregistry.md [hunk 10]
```diff
@@ -24,13 +24,33 @@ Now we have `uiRegistry` instance in the `registry` variable. We can use it to g
 var component = registry.get('%componentName%');
 ```
 
-For example:
+The `uiRegistry` instance allows to search the component by some property. 
+If you know some unique property of the component that you need to find, you can use the following code to get this component:
```
Comment (jeff-matthews) [inline guides/v2.2/ui_comp_guide/concepts/ui_comp_uiregistry.md]:
```suggestion
If you know a unique property of a component that you need to find, you can use the following code to get the component:
```

File: guides/v2.2/ui_comp_guide/concepts/ui_comp_uiregistry.md [hunk 11]
```diff
@@ -24,13 +24,33 @@ Now we have `uiRegistry` instance in the `registry` variable. We can use it to g
...
+
+```js
+var component = registry.get('%property% = %propertyValue%');
+```
+
+**Example of usages**
```
Comment (jeff-matthews) [inline guides/v2.2/ui_comp_guide/concepts/ui_comp_uiregistry.md]:
```suggestion
**Examples**
```

File: guides/v2.2/ui_comp_guide/concepts/ui_comp_uiregistry.md [hunk 12]
```diff
@@ -24,13 +24,33 @@ Now we have `uiRegistry` instance in the `registry` variable. We can use it to g
...
+
+The code below lists how to get a component by a full component name:
```
Comment (jeff-matthews) [inline guides/v2.2/ui_comp_guide/concepts/ui_comp_uiregistry.md]:
```suggestion
The following code shows how to get a component by a full component name:
```

File: guides/v2.2/ui_comp_guide/concepts/ui_comp_uiregistry.md [hunk 13]
```diff
@@ -24,13 +24,33 @@ Now we have `uiRegistry` instance in the `registry` variable. We can use it to g
...
 
 ```js
 // Admin > Products > Catalog > Add Product
 var fieldName = registry.get('product_form.product_form.product-details.container_name.name');
 ```
 
+The code below lists how to get a component by some property value:
```
Comment (jeff-matthews) [inline guides/v2.2/ui_comp_guide/concepts/ui_comp_uiregistry.md]:
```suggestion
The following code shows how to get a component by a property value:
```

File: guides/v2.2/ui_comp_guide/concepts/ui_comp_uiregistry.md [hunk 14]
```diff
@@ -24,13 +24,33 @@ Now we have `uiRegistry` instance in the `registry` variable. We can use it to g
...
+
+```js
+// Admin > Products > Catalog > Add Product
+var fieldName = registry.get('index = name');
+// or 
+fieldName = registry.get('inputName = product[name]');
+```
+
+By this code, you will obtain the same JS component as with using the full component name.
```
Comment (jeff-matthews) [inline guides/v2.2/ui_comp_guide/concepts/ui_comp_uiregistry.md]:
```suggestion
The previous example gets the same JS component as using the full component name.
```

**Comment by serhiyzhovnir** [comment]
@jeff-matthews We have added the changes according to editorial suggestions.
Thank you!

**Pull request merged**